### ISSUE TITLE
[master ticket] Integrating MapRoulette in other Editors

### ISSUE THREAD

**Issue by tordans** [maproulette/maproulette3#1737]
**The general idea is, to integrate MapRoulette in other editors.**
**This ticket is meant to track the status of discussions within other tickets.**
Please create separate tickets for discussions.

<table>
<thead>
<th>Editor</th>
<th>Tickets</th>
<th>Status</th>
<th>Main UseCase |</th>
</thead>
<tbody>
<tr>
<th>

**GoMap**

</th>
<td>

  * https://github.com/bryceco/GoMap/issues/240
  * MR https://github.com/osmlab/maproulette3/issues/1108

</td>
<td>zzZ; (?)</td>
<td>Solve challenges with local knowledge while walking down the street. Organized efforts by local community (of a city; district) but also discovery of other challenges that happen to be around the corner."

</td>
</tr>
<th>

**StreetComplete** SC

</th>
<td>

* Website: https://streetcomplete.app/
* [Search MR in SC Repo](https://github.com/streetcomplete/StreetComplete/search?q=maproulette&type=issues); _no focussed ticket, yet_ 
* MR _no focussed ticket, yet_</td>

<td>tbd</td>
<td>Same as GoMap

</td>
<tr>
<th>

**MapComplete** MC

</th>
<td>

  * [Search MR in SC Repo](https://github.com/pietervdvn/MapComplete/search?q=maproulette); _no focussed ticket, yet_
  * MR _no focussed ticket, yet_

</td>
<td>tbd</td>
<td>Same as GoMap; Interaction of themes is different. Might be a MC-theme for all MR challenges; might be and integration of specific MR challenges in specific MC-themes; might be a separate ("layer") from the themes…

MC does have a review-and-import-process based on special OSM notes already ([Blogpost](https://www.openstreetmap.org/user/Pieter%20Vander%20Vennet/diary/398782#comment52733)). That process could maybe be updated to use MapRoulette.

</td>
</tr>
<tr>
<th>

**iD**

</th>
<td>

  * [Search MR in iD Repo](https://github.com/openstreetmap/iD/search?q=maproulette&type=issues)
  * MR _no focussed ticket, yet_

</td>
<td>zzZ; (?)</td>
<td>Same workflow and use case as with Notes- and QA-Layers in iD: MR data can be made visible and edited/solved from within iD while editing a specific area.

</td>
</tr>
<tr>
<th>

**RapID**

</th>
<td>

  * Ticket https://github.com/facebook/Rapid/issues/1150
  * PR https://github.com/facebook/Rapid/pull/1221

</td>
<td>Actively being worked on</td>
<td>Right now the focus seems to be on getting the MR data into Rapid. There might be potential to go further to improve the editing experience.</td>
</tr>
<tr>
<th>

**JOSM**

</th>
<td>

  * JOSM Repo: ?
  * MR: ?

</td>
<td>?</td>
<td>Same as iD

</td>
</tr>
<tr>
<th>

**Vespucci**

</th>
<td>

  * Vespucci Repo: https://github.com/MarcusWolschon/osmeditor4android/issues/690
  * MR: Prerequisites for full support are https://github.com/maproulette/maproulette2/issues/476 and https://github.com/osmlab/maproulette3/issues/955

</td>
<td>Implemented but limited</td>
<td>Same as GoMap

</td>
</tr>
<tr>
<th>

**MapRoulette** MR mobile

</th>
<td>

* MR was not discussed, yet (Github Repo search)

</td>
<td>?</td>
<td>Same as StreetComplete and GoMap. But EveryDoor is Cross Plattform (Android + iPhone).</td>
</tr>
<tr>
<th>

**EveryDoor**

</th>
<td>

* Github: https://github.com/Zverik/every_door/
* Website: https://every-door.app/
* MR https://github.com/osmlab/maproulette3/issues/856

</td>
<td>?</td>
<td>Using MR as a separate app to solve local tickets; needs integration with GoMap and Vespucci to solve the given tasks; or only for quick-edit-challenges that do not require an external editor.</td>
</tr>
</tbody>
</table>

---

### Updates to this ticket

* 2022-06-22: MapComplete: [Add comment and link](https://www.openstreetmap.org/user/Pieter%20Vander%20Vennet/diary/398782#comment52733)
* 2023-02-10: Add EveryDoor.
* 2023-12-25: Split iD <> Rapid because there is active work to integrate into Rapid 🥳

**Comment by mvexel** [comment]
Vespucci does already have MR integration, see https://github.com/MarcusWolschon/osmeditor4android/issues/690

**Comment by joostschouppe** [comment]
Thinking more about features than specific editors, it would be nice to have a mobile app showing tasks that are "too hard, can't see" tasks.

**Comment by mvexel** [comment]
See also https://lists.openstreetmap.org/pipermail/talk-ca/2022-May/010428.html - someone is seeking a StreetComplete equivalent for iOS.

**Comment by mvexel** [comment]
@joostschouppe Agree. The idea being that some tasks could be done remotely and some would need verification in the field.

There is already a flag you can set when you create a challenge indicating that a challenge needs local knowledge. This could be used as a filter in mobile apps that want to integrate MapRoulette challenges. With some work we could enable this flag being attached to a task as well.

**Comment by mvexel** [comment]
The Vespucci integration is currently limited by https://github.com/osmlab/maproulette3/issues/1737 and https://github.com/maproulette/maproulette2/issues/476 as mentioned here: https://github.com/osmlab/maproulette3/issues/955

**Comment by mvexel** [comment]
This is not about editor integrations strictly, but I am starting to think more deeply about integrating with other QA signal sources. I wrote up a page about Mapillary [here](https://github.com/maproulette/maproulette3/wiki/Mapillary-Integration).

**Comment by tordans** [comment]
Update: I added EveryDoor to the list of possible integrations. This is also [a ping to Ilya's ticket on the question of funding features](https://github.com/Zverik/every_door/issues/539).

**Comment by pietervdvn** [comment]
Since this summer, MapComplete has some MapRoulette support.

First of all, you can see [all tasks in MapRoulette on MapComplete](mapcomplete.osm.be/maproulette).

Furthermore, it is possible to use a maproulette-challenge to do a guided import (to add new points or to add tags to an already existing points); but this needs some preparation to setup for thematic that want to do imports. I'm busy documenting the process [here](https://github.com/pietervdvn/MapComplete/blob/develop/Docs/Integrating_Maproulette.md)

**Comment by mvexel** [comment]
@tordans would it make sense to distill this into a MapRoulette project board?

**Comment by tordans** [comment]
Update/FYI: I just split the table lines for "iD/Rapid" since there is [an active PR in Rapid](https://github.com/facebook/Rapid/pull/1221) working on an integration.

**Comment by bhousel** [comment]
Update on this, as of Rapid v2.3 we've built a layer to allow users to work with MapRoulette tasks directly from Rapid.
https://github.com/facebook/Rapid/blob/main/CHANGELOG.md#230

Some more details on social media:

- Twitter/𝕏:  https://twitter.com/bhousel/status/1790060093733499259
- Mastodon:  https://mastodon.social/@bhousel/112434802439881843
- Threads:  https://www.threads.net/@houselb/post/C66uo9yRUaQ

**Comment by pietervdvn** [comment]
I'd like to bump https://github.com/maproulette/maproulette-backend/issues/1037 which would make my life easier

**Comment by mvexel** [comment]
This is now captured [on the wiki](https://github.com/maproulette/maproulette3/wiki/Integrating-MapRoulette-in-other-Editors)
